File: Leetcode/June Challenge/day 9/Is Subsequence.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        if (t.size()<s.size())
            return false;
        if (s.size()==0)
            return true;
        int j=0; 
        for(int i=0;i<t.size();i++)
        {
            if(t[i]==s[j])
            {
                j++;
                if (s.size()==j)
                    return true;
            }
        }
        return false;
    }
};

File: Leetcode/June Challenge/day 9/Is Subsequence.java
class Solution {
    public boolean isSubsequence(String s, String t) {
        if (t.length()<s.length())
            return false;
        if (s.length()==0)
            return true;
        int j=0; 
        for(int i=0;i<t.length();i++)
        {
            if(t.charAt(i)==s.charAt(j))
            {
                j++;
                if (s.length()==j)
                    return true;
            }
        }
        return false;
    }
}